File: Gombang/Weekly Contest 403/3194_Minimum_Average_of_Smallest_and_Largest_Elements.cs
using System;

public class Solution
{
    public double MinimumAverage(int[] nums)
    {
        // 오름차순 정렬
        Array.Sort(nums);

        double result = double.MaxValue;
        int left = 0;
        int right = nums.Length - 1;

        while (left < right)
        {
            double average = (nums[left] + nums[right]) / 2.0;
            if (result > average)
                result = average;

            left++;
            right--;
        }

        return result;
    }
}

File: Seori/Weekly Contest 403/3194. Minimum Average of Smallest and Largest Elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        # nums를 정렬하여 앞뒤로부터 인덱스를 이동하며 평균값을 구하고, 그 중 최소값을 반환한다. 
        # Sort nums and find the average values by moving the indexes from the front and back, and return the minimum value.
        nums.sort()
        averages = []
        n = len(nums)
        for i in range(n//2):
            averages.append((nums[i] + nums[n-1-i])/2)
        return min(averages)